### PR TITLE
Bug fix

### LINKED ISSUE #209: Bug buy/sell transaction
1. Quantity can input non-positive integers

### LINKED ISSUE #210: Bug sort command
1. Sorting stops displaying the remarks of the client.

### PR DESCRIPTION
**Pull request by Jo3LW**
1) Fix UI bug for sort command
2) Fix bug for user input for buy/sell command 
3) Improve UI for remark

### CODE REVIEW

File: src/main/java/seedu/address/model/transaction/Price.java [hunk 1]
```diff
@@ -20,22 +23,32 @@ public class Price {
...
         this.price = price;
     }
 
     /**
      * Returns true if a given string is a valid price.
      */
     public static boolean isValidPrice(String test) {
+        requireNonNull(test);
         boolean isDouble = true;
+        double num = 0.00;
```
Comment (jicsontoh) [inline src/main/java/seedu/address/model/transaction/Price.java]: Maybe its better to keep the original code?

**Review by jicsontoh — APPROVED** [review]
LGTM

File: src/main/java/seedu/address/model/transaction/Price.java (hunk 1 above)
Comment (Jo3LW) [reply src/main/java/seedu/address/model/transaction/Price.java]: Yeap thanks for pointing it out :)

File: src/main/java/seedu/address/logic/commands/RemarkCommand.java [hunk 2]
```diff
@@ -26,7 +26,7 @@ public class RemarkCommand extends Command {
             + "Example: " + COMMAND_WORD + " 2 "
             + "Fast buyer";
 
-    public static final String MESSAGE_SUCCESS = "New Remark created: %1$s\nLinked to Client: %2$s";
+    public static final String MESSAGE_SUCCESS = "New Remark created: %1$s\nLinked to Client: %2$s; Email: %3$s";
```
Comment (notle1706) [inline src/main/java/seedu/address/logic/commands/RemarkCommand.java:29]: Is email needed here?

File: src/main/java/seedu/address/logic/parser/ParserUtil.java [hunk 3]
```diff
@@ -181,9 +192,25 @@ public static Price parsePrice(String price) throws ParseException {
...
+        if (!Quantity.isValidQuantity_regex(trimmedQuantity)) {
+            throw new ParseException(Quantity.MESSAGE_CONSTRAINTS);
+        }
+        if (!Quantity.isSmallQuantity(trimmedQuantity)) {
+            throw new ParseException(Quantity.MESSAGE_CONSTRAINTS_LARGE);
+        }
+        if (!Quantity.isValidQuantity_nonZero(trimmedQuantity)) {
+            throw new ParseException(Quantity.MESSAGE_CONSTRAINTS_ZERO);
+        }
+
```
Comment (notle1706) [inline src/main/java/seedu/address/logic/parser/ParserUtil.java]: Good display of defensive programming.

File: src/main/java/seedu/address/logic/commands/RemarkCommand.java (hunk 2 above)
Comment (ElginL) [inline src/main/java/seedu/address/logic/commands/RemarkCommand.java:29]: I dont think email is needed here?

File: src/main/java/seedu/address/logic/commands/RemarkCommand.java [hunk 4]
```diff
@@ -68,7 +68,8 @@ public CommandResult execute(Model model) throws CommandException {
         editedClient.addRemark(remark);
         model.setClient(clientToEdit, editedClient);
 
-        return new CommandResult(String.format(MESSAGE_SUCCESS, remark.getText(), editedClient));
+        return new CommandResult(String.format(MESSAGE_SUCCESS, remark.getText(), editedClient.getName(),
+                editedClient.getEmail()));
```
Comment (notle1706) [inline src/main/java/seedu/address/logic/commands/RemarkCommand.java:72]: Is the email really necessary?

File: src/main/java/seedu/address/model/transaction/Price.java [hunk 5]
```diff
@@ -9,33 +9,75 @@
...
+     * Returns true if a give string is a valid positive price.
+     */
+    public static boolean isPositivePrice(String test) {
+        return !test.contains("-");
+    }
+
+    /**
+     * Returns true of a given string is not empty.
+     */
+    public static boolean isValidPrice_empty(String test) {
```
Comment (ElginL) [inline src/main/java/seedu/address/model/transaction/Price.java]: Why did you use `SNAKECASE` here when you used `CAMELCASE` everywhere else

File: src/main/java/seedu/address/model/transaction/Price.java [hunk 6]
```diff
@@ -9,33 +9,75 @@
...
+
+    /**
+     * Returns true if a give price is a less than 1 million.
+     */
+    public static boolean isSmallPrice(String test) {
+        requireNonNull(test);
+
+        boolean isSmall;
+
+        isSmall = Double.parseDouble(test) < 1000000;
```
Comment (ElginL) [inline src/main/java/seedu/address/model/transaction/Price.java]: Less than equals to here? Your error message says not more than 1 million

File: src/main/java/seedu/address/model/transaction/Price.java [hunk 7]
```diff
@@ -9,33 +9,75 @@
...
+
+        return isSmall;
```
Comment (ElginL) [inline src/main/java/seedu/address/model/transaction/Price.java]: from line 74 to 78, you can just do `return Double.parseDouble(test) < 1000000`, don't need to write unnecessary code.

File: src/main/java/seedu/address/model/transaction/Quantity.java [hunk 8]
```diff
@@ -7,29 +7,101 @@
...
+
     public static final String VALIDATION_REGEX = "\\d{1,}";
+
+    public static final String MESSAGE_CONSTRAINTS =
+            "Quantity should only contain numbers and no decimal point.";
+
+    public static final String MESSAGE_CONSTRAINTS_ZERO =
+            "Quantity should not be 0.";
+    public static final String MESSAGE_CONSTRAINTS_POSITIVE =
+            "Quantity should be not be negative.";
```
Comment (notle1706) [inline src/main/java/seedu/address/model/transaction/Quantity.java]: Perhaps this should be `should not be` instead.

File: src/main/java/seedu/address/model/transaction/Quantity.java [hunk 9]
```diff
@@ -7,29 +7,101 @@
...
+
+    public static final String MESSAGE_CONSTRAINTS_LARGE =
+            "Quantity should be not be more than 1 million.";
```
Comment (notle1706) [inline src/main/java/seedu/address/model/transaction/Quantity.java]: Same for this. Also, is 1 million large enough?

File: src/main/java/seedu/address/model/transaction/Quantity.java [hunk 10]
```diff
@@ -7,29 +7,101 @@
...
+    }
+
+
+    /**
+     * Returns true of a given string is non-zero.
+     */
+    public static boolean isValidQuantity_nonZero(String test) {
+        requireNonNull(test);
+        return Integer.parseInt(test) != 0;
+    }
```
Comment (ElginL) [inline src/main/java/seedu/address/model/transaction/Quantity.java]: are these method names snake case or camel case?

File: src/main/java/seedu/address/model/transaction/Quantity.java [hunk 11]
```diff
@@ -7,29 +7,101 @@
...
+            isInteger = false;
+        }
+
+        return isInteger;
+    }
+
+    /**
+     * Returns true of a given string fits the regex.
+     */
+    public static boolean isValidQuantity_regex(String test) {
```
Comment (notle1706) [inline src/main/java/seedu/address/model/transaction/Quantity.java]: Please keep to camelCase for method names.

File: src/main/java/seedu/address/model/transaction/Quantity.java [hunk 12]
```diff
@@ -7,29 +7,101 @@
...
+        requireNonNull(test);
         return test.matches(VALIDATION_REGEX);
     }
```
Comment (ElginL) [inline src/main/java/seedu/address/model/transaction/Quantity.java]: Is this supposed to be camel case or snake case?

File: src/main/java/seedu/address/model/transaction/Quantity.java [hunk 13]
```diff
@@ -7,29 +7,101 @@
...
+        this.quantity = Integer.toString(Integer.parseInt(quantity));
     }
 
     /**
      * Returns true if a given string is a valid quantity.
      */
+
     public static boolean isValidQuantity(String test) {
+        requireNonNull(test);
+        boolean isInteger = true;
```
Comment (ElginL) [inline src/main/java/seedu/address/model/transaction/Quantity.java]: You don't need isInteger, just return false at line 54 and return true at line 57

**Review by ElginL — COMMENTED** [review]
Good work, just a little bit of improvement will do

**Review by notle1706 — CHANGES_REQUESTED** [review]
Some changes to make, also some suggestions to consider.

File: src/main/java/seedu/address/model/transaction/Quantity.java [hunk 14]
```diff
@@ -3,33 +3,152 @@
...
+    public static final String MESSAGE_CONSTRAINTS_ZERO =
+            "Quantity should not be 0.";
+    public static final String MESSAGE_CONSTRAINTS_POSITIVE =
+            "Quantity should not be be negative.";
+
+    public static final String MESSAGE_CONSTRAINTS_LARGE =
+            "Quantity should be less than 1 million.";
+
+    public static final String MESSAGE_CONSTRAINTS_EMPTY = "Quantity should not be left empty.";
+
```
Comment (notle1706) [inline src/main/java/seedu/address/model/transaction/Quantity.java]: Perhaps you could clean up this portion of the code if you're not using it anymore?

**Review by notle1706 — APPROVED** [review]
LGTM, take a look at the comments

**Pull request merged**